### PR TITLE
Diary

### PR DESCRIPTION
**Pull request by JustynaSmoliga**

### CODE REVIEW

**Review by rosik1717 — CHANGES_REQUESTED** [review]

File: src/client/mealsClient.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+interface MealDto {
+  id: string;
+  mealType: MealType;
+  mealProducts: MealProductDto[];
+}
+
+export async function getMeals(date: string): Promise<DayMeals> {
+  const response = await axios.get(`http://localhost:8080/meals/${date}`, {});
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply src/client/mealsClient.ts:25]: Done

File: src/client/productClient.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+
+export interface Product {
+  id: string;
+  name: string;
+  totalCalories: number;
+  caloriesInGrams: number;
+  quantity: number;
+}
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply src/client/productClient.ts:9]: I will change it in next task.

File: src/features/meal/Meal.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import AddMealProduct from "../add-meal-product/AddMealProduct";
+import { MealProduct } from "../../slice/mealsSlice";
+
+interface MealProps {
+  title: string;
+  productsEaten: MealProduct[];
+  date: string;
+}
+
+export interface Product {
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply src/features/meal/Meal.tsx:14]: Ok

File: src/features/meal/Meal.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  id: number;
+  name: string;
+  totalCalories: number;
+  caloriesInGrams: number;
+  quantity: number;
+}
+
+const Meal: React.FC<MealProps> = (props) => {
+  const [addMealProductShowed, setAddMealProductShowed] = useState(false);
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply src/features/meal/Meal.tsx:23]: Done

File: src/features/meals/Meals.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import { RootState } from "../../app/store";
+import { fetchMeals } from "../../slice/mealsSlice";
+import Meal from "../meal/Meal";
+
+const Meals = () => {
+  const dispatch = useDispatch();
+  const dateIsoFormat = useSelector(
+    (state: RootState) => state.datePicker.diaryDate
+  );
+  const date = moment(dateIsoFormat).format("DD-MM-YYYY");
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply src/features/meals/Meals.tsx:14]: Done

**Pull request merged**